HatchGenerator/Freecad_Hatch_Generator.py

```diff
@@ -0,0 +1,4355 @@
+        cText = self.patternTypeFilterCombo.currentText()
+        filtered = self.filterObjects(sText, cText)
+        self.customPatternCombo.clear()
+        self.customPatternCombo.addItem("")
+        for (name, category, obj) in filtered:
+            self.customPatternCombo.addItem(name)
+
+    def _update_pattern_controls_visibility(self, show_builtin=True):
+        self.builtinPatternLabel.setVisible(show_builtin)
+        self.builtinPatternCombo.setVisible(show_builtin)
+        show_custom = not show_builtin
+        self.customPatternLabel.setVisible(show_custom)
+        self.customPatternCombo.setVisible(show_custom)
+        self.patternSearchLabel.setVisible(show_custom)
+        self.patternSearchField.setVisible(show_custom)
+        self.patternTypeFilterLabel.setVisible(show_custom)
+        self.patternTypeFilterCombo.setVisible(show_custom)
+
+        # Also show/hide the Additional Pattern Objects list if user picks "Custom"
+        self.patternObjectsLabel.setVisible(show_custom)
+        self.patternObjectsList.setVisible(show_custom)
+
+    def populateMultiList(self, listWidget):
+        listWidget.clear()
+        for (nm, cat, obj) in self.masterObjectList:
+            listWidget.addItem(nm)
+
+    def getSelectedObjectsFromList(self, listWidget):
+        selectedItems = listWidget.selectedItems()
+        selectedObjs = []
+        for it in selectedItems:
+            objName = it.text()
+            fcObj = self.doc.getObject(objName)
+            if fcObj:
+                selectedObjs.append(fcObj)
+        return selectedObjs
+
+    def getBaseSpacingInMM(self):
+        q = self.spacingInput.property("quantity")
+        return q.Value if q else 0.0
+
+    def onPreview(self):
+        """
+        Alternative Preview approach:
+        1) Create a temporary parametric hatch object with the user’s current property settings.
+        2) Recompute the doc, so the param object’s .Shape is generated.
+        3) Copy that shape into a simple Part::Feature named 'HatchPreview'.
+        4) Delete the temporary param object from the doc, leaving only the Part::Feature for visual preview.
+        """
+
+        from PySide import QtCore, QtWidgets
+        import Part
+
+        # 1) Remove any old preview shape
+        if self.current_temp_preview_name:
+            oldPreview = self.doc.getObject(self.current_temp_preview_name)
+            if oldPreview:
+                self.doc.removeObject(oldPreview.Name)
+        self.current_temp_preview_name = None
+
+        doc = self.doc
+        if not doc:
+            QtWidgets.QMessageBox.warning(self, "Error", "No active document found.")
+            return
+
+        baseName = self.baseCombo.currentText().strip()
+        if not baseName:
+            QtWidgets.QMessageBox.warning(self, "Error", "No base object selected.")
+            return
+
+        # 2) Create a brand new parametric object
+        tempName = "TempHatchParamObj"
```

Comment:
Why not create a temporary document for this (`tmp_doc = FreeCAD.newDocument(hidden=True, temp=True)` and `FreeCAD.closeDocument(tmp_doc.Name)`)?